### ISSUE TITLE
Can't get this to work in a package

### ISSUE THREAD

**Issue by clayne11** [gadicc/meteor-react-hotloader#19]
I have a simple package with two files. No global variables are referenced within this package. I import one component into the other. This package is then imported into another package (which does have global imports).

Do there have to be no "global" imports at all in my code for this to work, or simply none within the package that I'm trying to update?

**Comment by clayne11** [comment]
I've now removed even the "global" variables from the package that is importing my file and hot-reloading still isn't working for me.

**Comment by clayne11** [comment]
I tried using `.js` and `.jsx` and this component is also a React class (`class extends React.Component`)

**Comment by gadicc** [comment]
Firstly, thanks for reading over the README so thoroughly.

The package support is brand new, I hadn't planned to support it, but I'm currently developing a package and the reload times were getting annoying :>  The way I'm currently using it is that a component from the package is used by the app, and that seems to be working well.

I haven't tried importing from another package yet, although it should work.  I think the best thing though is if it's possible for you to share your code in a repo, so I can take a look at this exact case.

**Comment by clayne11** [comment]
My app is all packages (around 115 that I have written myself) since up until 1.3 that was the only way to control load order. I have packages importing into other packages all over the place.

Unfortunately I can't post my repo as it's my company's code base. I have a deadline coming up on Monday but I will try to create a reproduction after that.

**Comment by gadicc** [comment]
Haha yeah I remember those days fondly ;)

Ok, great!  I'll try test this scenario my side too over the weekend.

**Comment by clayne11** [comment]
It looks like this isn't working at all in my project and I can't figure out why. If I create an `import` folder at the top level of my app, then a `client` folder, and I create a simple class component in each of them, then import the `import` component into the `client` component, modifying either of them refreshes my entire app.

**Comment by clayne11** [comment]
I think my issue was with the `static-html` package.

I removed it and HMR is now working! Amazing!

See this issue: https://github.com/meteor/meteor/issues/6812

Now if only I could get your newest `fast` branch to work.

**Comment by clayne11** [comment]
Holy crap. This is incredible! Thank you so much. The I can't believe how fast it is. Literally instantaneous. WOW!

You're going to save me DAYS of waiting for rebuilds. This is amazing.

**Comment by gadicc** [comment]
Wait, so, your issues with `static-html` existed even before / without ecmascript-hot?  If the issue is only present with ecmascript-hot too, it's something I should look at.  Otherwise yeah, if there is any error in any part of Meteor ever, it can affect hotloading.

I'm really really happy this is finally working though... and from your positive experience so far.  I might have to stick that in the README at some point as a quote if that's ok :)  This was my whole motivation for writing the package, taking time off work now to speed up work in the future, and yeah, it's made a huge difference for me too.  Especially for working on actions too, depending on how you import them into your components.

Will track the other issue there... it's only a noticeable speed improvement on `> fast.3` if you have a lot of non-client-only files, but there have been a number of other improvements too (you can see the History.md).

**Comment by clayne11** [comment]
The issue with `static-html` (and `templating` for that matter) are issues with Meteor `1.3.2-rc.6`, whether I use the regular `ecmascript` package or `ecmascript-hot`. When I figured out that `static-html` was causing crashes in Meteor `1.3.2-rc.6` I checked out my `1.3.1` commit, removed `static-html`, and all of the sudden hot-reloading was working.

Something is up with the `static-html` package in my project. I'm still having issues running any version of `ecmascript-hot` beyong `1.3.1-fast.3` though, but at least I'm partway to a solution.
